src/main/java/com/google/api/codegen/discogapic/transformer/java/JavaDiscoGapicSurfaceTransformer.java

```diff
@@ -138,39 +193,527 @@ private StaticLangFileView generateApiFile(DiscoGapicInterfaceContext context) {
+    SurfaceNamer namer = context.getNamer();
+    SettingsDocView.Builder settingsDoc = SettingsDocView.newBuilder();
+    ProductServiceConfig productServiceConfig = new ProductServiceConfig();
+
+    // TODO(andrealin): Service address.
+    settingsDoc.serviceAddress("Service address.");
+    settingsDoc.servicePort(productServiceConfig.getServicePort());
+    settingsDoc.exampleApiMethodName(exampleApiMethod.name());
+    settingsDoc.exampleApiMethodSettingsGetter(exampleApiMethod.settingsGetterName());
+    settingsDoc.apiClassName(namer.getApiWrapperClassName(context.getInterfaceConfig()));
+    settingsDoc.settingsVarName(namer.getApiSettingsVariableName(context.getInterfaceConfig()));
+    settingsDoc.settingsClassName(namer.getApiSettingsClassName(context.getInterfaceConfig()));
+    settingsDoc.settingsBuilderVarName(
+        namer.getApiSettingsBuilderVarName(context.getInterfaceConfig()));
+    settingsDoc.hasDefaultInstance(context.getInterfaceConfig().hasDefaultInstance());
+    return settingsDoc.build();
+  }
+
+  private List<StaticLangApiMethodView> generateApiMethods(DiscoGapicInterfaceContext context) {
+    List<StaticLangApiMethodView> apiMethods = new ArrayList<>();
+
+    for (MethodModel method : context.getSupportedMethods()) {
+      MethodConfig methodConfig = context.getMethodConfig(method);
+      DiscoGapicMethodContext requestMethodContext = context.asRequestMethodContext(method);
+
+      if (methodConfig.isPageStreaming()) {
+        if (methodConfig.isFlattening()) {
+          for (FlatteningConfig flatteningGroup : methodConfig.getFlatteningConfigs()) {
+            DiscoGapicMethodContext flattenedMethodContext =
+                context.asFlattenedMethodContext(
+                    method, flatteningGroup, context.getInterfaceName());
+            apiMethods.add(
+                apiMethodTransformer.generatePagedFlattenedMethod(flattenedMethodContext));
+          }
+        }
+        apiMethods.add(apiMethodTransformer.generatePagedRequestObjectMethod(requestMethodContext));
+        apiMethods.add(apiMethodTransformer.generatePagedCallableMethod(requestMethodContext));
+        apiMethods.add(
+            apiMethodTransformer.generateUnpagedListCallableMethod(requestMethodContext));
+      } else {
+        if (methodConfig.isFlattening()) {
+          for (FlatteningConfig flatteningGroup : methodConfig.getFlatteningConfigs()) {
+            DiscoGapicMethodContext flattenedMethodContext =
+                context.asFlattenedMethodContext(
+                    method, flatteningGroup, context.getInterfaceName());
+            apiMethods.add(apiMethodTransformer.generateFlattenedMethod(flattenedMethodContext));
+          }
+        }
+        apiMethods.add(apiMethodTransformer.generateRequestObjectMethod(requestMethodContext));
+        apiMethods.add(apiMethodTransformer.generateCallableMethod(requestMethodContext));
+      }
+    }
+
+    return apiMethods;
+  }
```

Comment:
No, I'd prefer not to follow that kind of pattern. Subclassing often makes code flows harder to follow. I'd prefer the two classes to be independent, and use utility methods in a third class.